modules/video/src/tracking/tracker_nano.cpp

```diff
@@ -0,0 +1,361 @@
+#endif
+
+typedef std::vector<int> MatShape;
+
+namespace cv {
+
+TrackerNano::TrackerNano()
+{
+    // nothing
+}
+
+TrackerNano::~TrackerNano()
+{
+    // nothing
+}
+
+TrackerNano::Params::Params()
+{
+    backbone = "./backbone.onnx";
+    neckhead = "./neckhead.onnx";
+#ifdef HAVE_OPENCV_DNN
+    backend = dnn::DNN_BACKEND_DEFAULT;
+    target = dnn::DNN_TARGET_CPU;
+#else
+    backend = -1;  // invalid value
+    target = -1;  // invalid value
+#endif
+}
+
+#ifdef HAVE_OPENCV_DNN
+static void softmax(const Mat& src, Mat& dst)
+{
+    Mat maxVal;
+    cv::max(src.row(1), src.row(0), maxVal);
+
+    src.row(1) -= maxVal;
+    src.row(0) -= maxVal;
+
+    exp(src, dst);
+
+    Mat sumVal = dst.row(0) + dst.row(1);
+    dst.row(0) = dst.row(0) / sumVal;
+    dst.row(1) = dst.row(1) / sumVal;
+}
+
+static float sizeCal(float w, float h)
+{
+    float pad = (w + h) * float(0.5);
+    float sz2 = (w + pad) * (h + pad);
+    return sqrt(sz2);
+}
+
+static Mat sizeCal(const Mat& w, const Mat& h)
+{
+    Mat pad = (w + h) * 0.5;
+    Mat sz2 = (w + pad).mul((h + pad));
+
+    cv::sqrt(sz2, sz2);
+    return sz2;
+}
+
+static void elementMax(Mat& srcDst)
+{
+    size_t totalV = srcDst.total();
+    float* ptr = srcDst.ptr<float>(0);
+    for (size_t i = 0; i < totalV; i++)
+    {
+        float val = *(ptr + i);
+        *(ptr + i) = std::max(val, 1.0f/val);
+    }
+}
+
+class TrackerNanoImpl : public TrackerNano
+{
+public:
+    TrackerNanoImpl(const TrackerNano::Params& parameters)
+        : params(parameters)
+    {
+        backbone = dnn::readNet(params.backbone);
+        neckhead = dnn::readNet(params.neckhead);
+
+        CV_Assert(!backbone.empty());
+        CV_Assert(!neckhead.empty());
+
+        backbone.setPreferableBackend(params.backend);
+        backbone.setPreferableTarget(params.target);
+        neckhead.setPreferableBackend(params.backend);
+        neckhead.setPreferableTarget(params.target);
+    }
+
+    void init(InputArray image, const Rect& boundingBox) CV_OVERRIDE;
+    bool update(InputArray image, Rect& boundingBox) CV_OVERRIDE;
+    float getTrackingScore() CV_OVERRIDE;
+
+    TrackerNano::Params params;
+
+protected:
+    dnn::Net backbone, neckhead;
+    Mat image;
+
+    struct trackerConfig
+    {
+        int exemplarSize = 127;
+        int instanceSize = 255;
+        float windowInfluence = 0.455f;
+        float lr = 0.37f;
+        float contextAmount = 0.5;
+        bool swapRB = true;
+        int totalStride = 16;
+        float penaltyK = 0.055f;
+
+        int scoreSize = (instanceSize - exemplarSize) / totalStride + 8;
+
+        Mat hanningWindow;
+        Mat grid2searchX, grid2searchY;
+        
+        Size imgSize = {0, 0};
+        
+        // Save the target bounding box for each frame.
+        std::vector<float> targetSz = {0, 0};  // H and W of bounding box
+        std::vector<float> targetPos = {0, 0}; // center point of bounding box (x, y)
+
+        float tracking_score;
+    };
```

Comment:
I propose to split the structure on two:
- TrackerConfig. Place fields like `exemplarSize`, `instanceSize`, etc. They control the tracker behavior, but do not change during the tracker lifetime. It should be public to give opportunity for customization.
- TrackerState. `tracking_score`, `targetPos`. It's internal tracker state and should stay private.

Reply:
Agree.

Reply:
Fixed.